src/stream/src/executor/source/source_executor.rs

```diff
@@ -122,6 +123,34 @@ impl<S: StateStore> SourceExecutor<S> {
             .iter()
             .map(|column_desc| column_desc.column_id)
             .collect_vec();
+
+        let (schema_change_tx, mut schema_change_rx) =
+            tokio::sync::mpsc::channel::<(SchemaChangeEnvelope, oneshot::Sender<()>)>(16);
+        let meta_client = self.actor_ctx.meta_client.clone();
+        // spawn a task to handle schema change event from source parser
+        let _join_handle = tokio::task::spawn(async move {
+            while let Some((schema_change, finish_tx)) = schema_change_rx.recv().await {
+                let table_names = schema_change.table_names();
+                tracing::info!("recv a schema change event for tables: {:?}", table_names);
+                // TODO: retry on rpc error
+                if let Some(ref meta_client) = meta_client {
+                    match meta_client
+                        .auto_schema_change(schema_change.to_protobuf())
+                        .await
+                    {
+                        Ok(_) => {
+                            tracing::info!("schema change success for tables: {:?}", table_names);
+                            finish_tx.send(()).unwrap();
+                        }
+                        Err(e) => {
+                            tracing::error!(error = ?e.as_report(), "schema change error");
+                            finish_tx.send(()).unwrap();
```

Comment:
I have added retry for the frontend client on meta, and plan to note down the failed schema event into event log in follow up PRs, so that user can manually execute the DDL if auto schema change fail.